department-of-veterans-affairs/caseflow
# APPEALS-30937 - ‘Days Waiting’ condition for report

Resolves [APPEALS-30937](https://jira.devops.va.gov/browse/APPEALS-30937)

# Description
Adds the Days Waiting component content

Adds validation and error messages to the conditions section and report type dropdown

## Acceptance Criteria
- [ ] Additional condition option: Days Waiting 
    - [ ] Drop-down options for ‘Time Range’ type include:
        - [ ] Less than
        - [ ] More than
        - [ ] Equal to
        - [ ] Between
    - [ ] One text box is included to decipher option for Less than, More than, Equal to option.
    - [ ] Two text boxes are included to decipher option for Between range. 


## Testing Plan

1. Log in as ACBAUERVVHAH, navigate to /decision_reviews/vha/report
2. Add a condition and select Days Waiting
3. For Time Range: Less than, More than, and Equal to should have one number field called 'Number of days'. Between should have two number fields for 'Min days' and 'Max days'


# Frontend
## User Facing Changes

<img width="1115" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/c15c3d41-d023-40f3-9e69-1c2f9f0e0854">
<img width="1085" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/17840614/84368f6d-7670-4949-965b-0034187fd9ea">


## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
